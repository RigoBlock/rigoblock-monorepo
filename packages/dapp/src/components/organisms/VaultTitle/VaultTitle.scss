@import "../../_settings/variables.scss";

.vault-title {
  .vault-title-header {
    display: flex;
    justify-content: space-between;
    align-content: flex-start;
    align-items: center;

    .title {
      text-transform: uppercase;
    }
  }
  .amount-wrapper {
    max-width: 2 * $dist-6xl;
  }
}
