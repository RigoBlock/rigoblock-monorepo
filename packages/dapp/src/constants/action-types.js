import keyMirror from 'keymirror'

export const actionTypes = keyMirror({
  // Global actions
  GLOBAL_INIT: null,
  // Blockchain actions
  BLOCKCHAIN_INIT: null,
  BLOCKCHAIN_ERROR: null,
  LOGGED_IN: null,
  LOGGED_OUT: null
})
