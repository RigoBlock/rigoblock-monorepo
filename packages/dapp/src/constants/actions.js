export default {
  persist: 'persist',
  form: '@@redux-form'
}
