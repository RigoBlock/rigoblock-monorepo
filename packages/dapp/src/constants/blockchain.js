export const VAULT = 'vault'
export const DRAGO = 'drago'
