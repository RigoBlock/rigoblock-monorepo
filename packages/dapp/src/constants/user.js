export default {
  MANAGER: 'manager',
  INVESTOR: 'investor'
}
