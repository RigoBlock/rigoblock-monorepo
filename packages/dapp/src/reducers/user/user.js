import { actionTypes } from '../../constants/action-types'
import CONSTANTS from '../../constants/user'
import moment from 'moment-timezone'
import persistentDecorator from '../../store/persistentDecorator'

const initialState = {
  timezone: moment.tz(moment.tz.guess()).format('Z'),
  type: CONSTANTS.INVESTOR,
  wallets: {}
}

function userReducer(state = initialState, action) {
  switch (action.type) {
    case actionTypes.LOGGED_IN:
      return {
        ...state,
        wallets: {
          [action.payload.origin]: action.payload.account
        }
      }
    case actionTypes.LOGGED_OUT:
      return {
        ...state,
        wallets: {}
      }
    case actionTypes.PREFERENCE_CHANGE:
      return {
        ...state,
        timezone: action.payload.timezone
      }
    default:
      return state
  }
}

export default persistentDecorator(userReducer, 'user')
