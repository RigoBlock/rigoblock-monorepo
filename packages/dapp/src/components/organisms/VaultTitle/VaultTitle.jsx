import './VaultTitle.scss'
import { ETH } from '../../../constants/blockchain'
import { ETH_TO_WEI } from '../../../constants/utils'
import { connect } from 'react-redux'
import Amount, { AMOUNT_SIZES } from '../../molecules/Amount'
import Button from '../../atoms/Button'
import CallToAction from '../../molecules/CallToAction'
import PropTypes from 'prop-types'
import React from 'react'
import Title, { TITLE_SIZES } from '../../atoms/Title'

let VaultTitle = ({ vaultAddress, vaults }) => {
  const selectedVault = vaults[vaultAddress]
  const vaultSupply = selectedVault.totalSupply.div(ETH_TO_WEI)
  return (
    <div className="vault-title">
      <div className="vault-title-header">
        <Title size={TITLE_SIZES.LARGE}>{selectedVault.symbol}</Title>
        <CallToAction>
          {/* TODO: implement logic to deposit or withdraw to/from vault */}
          <Button onClick={() => {}}>Deposit</Button>
          <Button onClick={() => {}}>Withdraw</Button>
        </CallToAction>
      </div>
      <div className="amount-wrapper">
        <Amount
          value={vaultSupply}
          symbol={ETH}
          valueSize={AMOUNT_SIZES.LARGE}
          symbolSize={AMOUNT_SIZES.SMALL}
        />
      </div>
    </div>
  )
}

VaultTitle.propTypes = {
  vaultAddress: PropTypes.string.isRequired,
  vaults: PropTypes.object.isRequired
}

VaultTitle = connect(state => {
  const { currentAccount } = state.preferences
  return {
    vaults: currentAccount
      ? state.blockChain.accounts[currentAccount].vaults
      : {}
  }
})(VaultTitle)

export default VaultTitle
