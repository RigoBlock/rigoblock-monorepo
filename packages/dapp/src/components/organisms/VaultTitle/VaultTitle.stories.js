import '../../_settings/_base.scss'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import React from 'react'
import VaultTitle from './VaultTitle'
import mockStore from '../../../fixtures/store'

const store = mockStore()
const props = { vaultAddress: '0x86a1ba4d485ce346bded508e2426798f825558be' }

storiesOf('Organisms/VaultTitle', module)
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <div style={{ width: '478px', background: 'white', padding: '16px' }}>
      {story()}
    </div>
  ))
  .addDecorator(story => <MemoryRouter>{story()}</MemoryRouter>)
  .addDecorator(story => <Provider store={store}>{story()}</Provider>)
  .add('default', () => <VaultTitle {...props} />)
