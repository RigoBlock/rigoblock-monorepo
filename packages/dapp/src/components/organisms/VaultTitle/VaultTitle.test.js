import { BigNumber } from 'bignumber.js'
import { INVESTOR, MANAGER } from '../../../constants/user'
import { mount } from 'enzyme'
import React from 'react'
import VaultTitle from './VaultTitle.jsx'
import toJson from 'enzyme-to-json'

const vault = {
  id: new BigNumber('0'),
  name: 'First Vault',
  symbol: 'asd',
  owner: '0x242b2dd21e7e1a2b2516d0a3a06b58e2d9bf9196',
  group: '0x7ce6e371085cb611fb46d5065397223ef2f952ff',
  totalSupply: new BigNumber('28640000000000000000'),
  transactionFee: new BigNumber('2.50')
}

describe('VaultTitle component', () => {
  it('renders correctly as manager', () => {
    expect(
      toJson(mount(<VaultTitle vault={vault} userType={INVESTOR} />))
    ).toMatchSnapshot()
  })

  it('renders correctly as investor', () => {
    expect(
      toJson(mount(<VaultTitle vault={vault} userType={MANAGER} />))
    ).toMatchSnapshot()
  })
})
