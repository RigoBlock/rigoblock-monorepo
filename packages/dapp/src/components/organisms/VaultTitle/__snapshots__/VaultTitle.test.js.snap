// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VaultTitle component renders correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [Function],
      "subscribe": [MockFunction],
    }
  }
>
  <Connect(VaultTitle)
    vaultAddress="0x86a1ba4d485ce346bded508e2426798f825558be"
  >
    <VaultTitle
      dispatch={[MockFunction]}
      vaultAddress="0x86a1ba4d485ce346bded508e2426798f825558be"
      vaults={
        Object {
          "0x421e1cef6e85e78da2470e54af64a626f45afb85": Object {
            "group": "0x7ce6e371085cb611fb46d5065397223ef2f952ff",
            "id": "1",
            "name": "Third Vault",
            "owner": "0x7328ef1d7ab7583eb9968b2f4a9c900f8a2e2d6d",
            "symbol": "das",
            "totalSupply": "21343170000000000000",
          },
          "0x86a1ba4d485ce346bded508e2426798f825558be": Object {
            "group": "0x7ce6e371085cb611fb46d5065397223ef2f952ff",
            "id": "0",
            "name": "First Vault",
            "owner": "0x242b2dd21e7e1a2b2516d0a3a06b58e2d9bf9196",
            "symbol": "asd",
            "totalSupply": "28640000000000000000",
          },
        }
      }
    >
      <div
        className="vault-title"
      >
        <div
          className="vault-title-header"
        >
          <Title
            className={null}
            size="large"
          >
            <div
              className="title large"
            >
              asd
            </div>
          </Title>
          <CallToAction>
            <div
              className="call-to-action"
            >
              <Button
                appearance="primary"
                onClick={[Function]}
                type="button"
              >
                <withInk(withTooltip(Button))
                  className="base-button"
                  flat={true}
                  inkTransitionEnterTimeout={450}
                  inkTransitionLeaveTimeout={300}
                  inkTransitionOverlap={150}
                  onClick={[Function]}
                  primary={true}
                  swapTheming={false}
                  type="button"
                >
                  <withTooltip(Button)
                    className="base-button"
                    flat={true}
                    ink={
                      <InkContainer
                        className={undefined}
                        disabledInteractions={undefined}
                        inkClassName={undefined}
                        inkStyle={undefined}
                        pulse={undefined}
                        style={undefined}
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}
                        waitForInkTransition={undefined}
                      />
                    }
                    onClick={[Function]}
                    primary={true}
                    swapTheming={false}
                    type="button"
                  >
                    <Button
                      className="base-button"
                      fixedPosition="br"
                      flat={true}
                      iconBefore={true}
                      ink={
                        <InkContainer
                          className={undefined}
                          disabledInteractions={undefined}
                          inkClassName={undefined}
                          inkStyle={undefined}
                          pulse={undefined}
                          style={undefined}
                          transitionEnterTimeout={450}
                          transitionLeaveTimeout={300}
                          transitionOverlap={150}
                          waitForInkTransition={undefined}
                        />
                      }
                      onClick={[Function]}
                      primary={true}
                      swapTheming={false}
                      type="button"
                    >
                      <button
                        className="md-btn md-btn--flat md-btn--text md-pointer--hover md-text--theme-primary md-ink--primary md-inline-block base-button"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                        type="button"
                      >
                        <InkContainer
                          key="ink-container"
                          transitionEnterTimeout={450}
                          transitionLeaveTimeout={300}
                          transitionOverlap={150}
                        >
                          <TransitionGroup
                            childFactory={[Function]}
                            className="md-ink-container"
                            component="div"
                          >
                            <div
                              className="md-ink-container"
                            />
                          </TransitionGroup>
                        </InkContainer>
                        Deposit
                      </button>
                    </Button>
                  </withTooltip(Button)>
                </withInk(withTooltip(Button))>
              </Button>
              <Button
                appearance="primary"
                onClick={[Function]}
                type="button"
              >
                <withInk(withTooltip(Button))
                  className="base-button"
                  flat={true}
                  inkTransitionEnterTimeout={450}
                  inkTransitionLeaveTimeout={300}
                  inkTransitionOverlap={150}
                  onClick={[Function]}
                  primary={true}
                  swapTheming={false}
                  type="button"
                >
                  <withTooltip(Button)
                    className="base-button"
                    flat={true}
                    ink={
                      <InkContainer
                        className={undefined}
                        disabledInteractions={undefined}
                        inkClassName={undefined}
                        inkStyle={undefined}
                        pulse={undefined}
                        style={undefined}
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}
                        waitForInkTransition={undefined}
                      />
                    }
                    onClick={[Function]}
                    primary={true}
                    swapTheming={false}
                    type="button"
                  >
                    <Button
                      className="base-button"
                      fixedPosition="br"
                      flat={true}
                      iconBefore={true}
                      ink={
                        <InkContainer
                          className={undefined}
                          disabledInteractions={undefined}
                          inkClassName={undefined}
                          inkStyle={undefined}
                          pulse={undefined}
                          style={undefined}
                          transitionEnterTimeout={450}
                          transitionLeaveTimeout={300}
                          transitionOverlap={150}
                          waitForInkTransition={undefined}
                        />
                      }
                      onClick={[Function]}
                      primary={true}
                      swapTheming={false}
                      type="button"
                    >
                      <button
                        className="md-btn md-btn--flat md-btn--text md-pointer--hover md-text--theme-primary md-ink--primary md-inline-block base-button"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                        type="button"
                      >
                        <InkContainer
                          key="ink-container"
                          transitionEnterTimeout={450}
                          transitionLeaveTimeout={300}
                          transitionOverlap={150}
                        >
                          <TransitionGroup
                            childFactory={[Function]}
                            className="md-ink-container"
                            component="div"
                          >
                            <div
                              className="md-ink-container"
                            />
                          </TransitionGroup>
                        </InkContainer>
                        Withdraw
                      </button>
                    </Button>
                  </withTooltip(Button)>
                </withInk(withTooltip(Button))>
              </Button>
            </div>
          </CallToAction>
        </div>
        <div
          className="amount-wrapper"
        >
          <Amount
            symbol="ETH"
            symbolSize="small"
            value={"28.64"}
            valueSize="large"
          >
            <div
              className="amount"
            >
              <div
                className="amount-symbol small"
              >
                ETH
              </div>
              <ItemValue
                currency="ETH"
                currencyGrowth={null}
                growth={null}
                itemValue={"28.64"}
                precision={2}
                valueSize="large"
              >
                <div
                  className="no-events"
                >
                  <span
                    className="item-value large"
                  >
                    28.64
                  </span>
                </div>
              </ItemValue>
            </div>
          </Amount>
        </div>
      </div>
    </VaultTitle>
  </Connect(VaultTitle)>
</Provider>
`;
